modules/product/src/main/java/com/opengamma/strata/product/SecurityId.java

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product;
+
+import java.io.Serializable;
+
+import org.joda.beans.PropertyDefinition;
+import org.joda.convert.FromString;
+import org.joda.convert.ToString;
+
+import com.opengamma.strata.basics.market.ReferenceData;
+import com.opengamma.strata.basics.market.ReferenceDataId;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.id.StandardId;
+
+/**
+ * An identifier for a security.
+ * <p>
+ * This identifier is used to obtain a {@link Security} from {@link ReferenceData}.
+ * <p>
+ * A security identifier uniquely identifies a security within the system.
+ * A real-world security will typically have multiple identifiers.
+ * The only restriction placed on the identifier is that it is sufficiently
+ * unique for the reference data lookup. As such, it is acceptable to use
+ * an identifier from a well-known global or vendor symbology.
+ */
+public final class SecurityId
+    implements ReferenceDataId<Security<?>>, Serializable {
+
+  /** Serialization version. */
+  private static final long serialVersionUID = 1L;
+
+  /**
+   * The identifier, expressed as a standard two-part identifier.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final StandardId standardId;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Obtains an instance from a scheme and value.
+   * <p>
+   * The scheme must be non-empty and match the regular expression '{@code [A-Za-z0-9:/+.=_-]*}'.
+   * This permits letters, numbers, colon, forward-slash, plus, dot, equals, underscore and dash.
+   * If necessary, the scheme can be encoded using {@link StandardId#encodeScheme(String)}.
+   * <p>
+   * The value must be non-empty and match the regular expression '{@code [!-z][ -z]*}'.
+   *
+   * @param scheme  the scheme of the identifier, not empty
+   * @param value  the value of the identifier, not empty
+   * @return the security identifier
+   * @throws IllegalArgumentException if the scheme or value is invalid
+   */
+  public static SecurityId of(String scheme, String value) {
+    return of(StandardId.of(scheme, value));
+  }
+
+  /**
+   * Creates an instance from a standard two-part identifier.
+   *
+   * @param standardId  the underlying standard two-part identifier
+   * @return the security identifier
+   */
+  public static SecurityId of(StandardId standardId) {
+    return new SecurityId(standardId);
+  }
+
+  /**
+   * Parses an {@code StandardId} from a formatted scheme and value.
+   * <p>
+   * This parses the identifier from the form produced by {@code toString()}
+   * which is '{@code $scheme~$value}'.
```

Comment:
I'm not sure whether the `$` signs here make it more or less obvious what's happening. I can see what you're saying, but I don't know how open to misinterpretation it is. Maybe add an example.